multinet/api/models/workspace.py

```diff
@@ -81,6 +81,32 @@ def get_user_permission(self, user: User) -> Optional[WorkspaceRole]:
         """Get the WorkspaceRole for a given user on this workspace."""
         return WorkspaceRole.objects.filter(workspace=self.pk, user=user.pk).first()
 
+    def get_user_permission_level(self, user: User) -> Optional[str]:
+        """
+        Get a the string label of a user's workspace role, or None.
+
+        This function differs from Workspace::get_user_permisssion. Whereas that
+        function returns a WorkspaceRole object describing a user's permission for
+        a particular workspace, this function returns a human-readable representation
+        of that permission level. Since ownership of a workspace is modeled by a property
+        on that workspace, this function takes that into account, and returns 'owner' if
+        the given user is the owner of this workspace. Moreover, even if a user does not
+        have explicit permission via the ownersip relationship or a WorkspaceRole object,
+        they are still considered a reader if this workspace is public. Therefore, if this
+        workspace is public and the user passed in has no explicit permission, 'reader' will
+        be returned.
+        """
+        if self.owner == user:
+            return 'owner'
+
+        workspace_role = self.get_user_permission(user)
+        if workspace_role is None:
+            if self.public:
+                return 'reader'
```

Comment:
Kind of related to my above suggestion, I think eliminating any more hardcoded permission strings would be ideal
```suggestion
                return WorkspaceRoleChoice.READER.label
```